### DISCUSSION TITLE
JSON output?

### DISCUSSION THREAD

**Discussion by lostmygithubaccount** [ml-explore/mlx#402] [General]
one of the most powerful uses w/ LLMs I've found is the ability to generate structured data (i.e. JSON/function calling output) from unstructured text. is this supported MLX? could it be? what would that look like?

**Comment by awni** [comment]
I think the right buzz word for this is "guided generation".

There has been an effort to do this with MLX LLMs: https://github.com/sacha-ichbiah/outlines-mlx

The Outlines package seems to be an actively maintained package for this, and if they could support MLX that might be the best path for this https://github.com/outlines-dev/outlines

Unlikely we would implement this in mlx core or mlx-examples as it's slightly outside the scope.
